runtime/src/impls.rs

```diff
@@ -867,6 +871,157 @@ where
 	}
 }
 
+pub struct DoughnutCallValidator;
+impl seed_pallet_common::ExtrinsicChecker for DoughnutCallValidator {
+	type Call = RuntimeCall;
+	type Extra = Topping;
+	type Result = DispatchResult;
+	fn check_extrinsic(call: &Self::Call, topping: &Self::Extra) -> DispatchResult {
+		// matcher to select the actual call to validate
+		let mut actual_call: Self::Call = call.clone();
+		match &call {
+			RuntimeCall::Futurepass(pallet_futurepass::Call::proxy_extrinsic {
+				call: inner_call,
+				..
+			}) => actual_call = *inner_call.clone(),
+			RuntimeCall::FeeProxy(pallet_fee_proxy::Call::call_with_fee_preferences {
+				call: inner_call_1,
+				..
+			}) => {
+				if let RuntimeCall::Futurepass(pallet_futurepass::Call::proxy_extrinsic {
+					call: inner_call_2,
+					..
+				}) = *inner_call_1.clone()
+				{
+					actual_call = *inner_call_2.clone();
+				} else {
+					actual_call = *inner_call_1.clone();
+				}
+			},
+			_ => actual_call = call.clone(),
+		}
```

Comment:
To remove the warning we can do something like this:
```suggestion
		let call: Self::Call = &call {
			RuntimeCall::Futurepass(pallet_futurepass::Call::proxy_extrinsic {
				call: inner_call,
				..
			}) => *inner_call.clone(),
			RuntimeCall::FeeProxy(pallet_fee_proxy::Call::call_with_fee_preferences {
				call: inner_call_1,
				..
			}) => {
				if let RuntimeCall::Futurepass(pallet_futurepass::Call::proxy_extrinsic {
					call: inner_call_2,
					..
				}) = *inner_call_1.clone()
				{
					*inner_call_2.clone();
				} else {
					*inner_call_1.clone();
				}
			},
			_ => call.clone(),
		}
```

Reply:
[ffd4629](https://github.com/futureversecom/trn-seed/pull/760/commits/ffd4629851906371c80514b54a79229c864f3b63)